src/AnyAddress.cpp

```diff
@@ -0,0 +1,36 @@
+// Copyright © 2017-2022 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "AnyAddress.h"
+
+#include <utility>
+#include "Coin.h"
+
+namespace TW {
+
+Data AnyAddress::getData() const {
+    Data data;
+    try {
+        data = TW::addressToData(coin, address);
+    } catch (...) {}
+    return data;
+}
+
+AnyAddress* AnyAddress::createAddress(const std::string& address, enum TWCoinType coin) {
+    auto normalized = TW::normalizeAddress(coin, address);
+    if (normalized.empty()) {
+        return nullptr;
+    }
+
+    return new AnyAddress(normalized, coin);
+}
+
+AnyAddress* AnyAddress::createAddress(const PublicKey& publicKey, enum TWCoinType coin) {
+    auto derivedAddress = TW::deriveAddress(coin, publicKey);
+    return new AnyAddress(derivedAddress, coin);
+}
```

Comment:
Here you can take string by copy and do the following:

(AnyAddress doesn't require a constructor)

```cpp
return new AnyAddress{.address = std::move(derivedAddress), .coin = coin};
```